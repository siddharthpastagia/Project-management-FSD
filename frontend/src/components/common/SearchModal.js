import React, { useEffect, useState } from "react";
import { Modal } from "react-bootstrap";
import Table from "./Table";

const SearchModal = props => {
  const columns = [{ dataField: "name", text: "Project Name" }];
  const onSelectItem = row => {
    props.onSearch(row);
    props.onCloseModal();
  };

  return (
    <Modal show={props.showModal} onHide={props.onCloseModal}>
      <Modal.Header closeButton>
        <Modal.Title>{props.heading}</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        {/* <div
          onClick={() => {
            onSelectItem(props.data[0]);
          }}
        > */}
        <Table
          columns={columns}
          data={props.data}
          keyField="id"
          clickedRow={onSelectItem}
        />

        {/* </div> */}
      </Modal.Body>
    </Modal>
  );
};
export default SearchModal;
