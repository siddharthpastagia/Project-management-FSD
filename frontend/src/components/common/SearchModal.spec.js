import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import SearchModal from "./SearchModal";

describe("Search Modal Component", () => {
  afterEach(cleanup);
  let showModal = true;
  const data = [{ projectName: "A" }, { projectName: "B" }];
  const columns = [{ dataField: "projectName", text: "Project List" }];
  const onCloseModal = jest.fn();
  const onSearch = jest.fn();
  it("Should render Search Modal component", () => {
    const { container } = render(<SearchModal />);
    expect(container).toMatchSnapshot();
  });

  it("should handle onSearch prop", () => {
    const { container, getByText } = render(
      <SearchModal
        showModal={showModal}
        data={data}
        columns={columns}
        onCloseModal={() => (showModal = false)}
        onSearch={() => onSearch(data[0].projectName)}
      />
    );
    const firstRow = getByText("A");
    fireEvent.click(firstRow);
    expect(container).toMatchSnapshot();
  });
});
