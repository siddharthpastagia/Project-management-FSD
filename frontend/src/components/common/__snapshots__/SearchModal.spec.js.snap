// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Modal Component Should render Search Modal component 1`] = `<div />`;

exports[`Search Modal Component should handle onSearch prop 1`] = `
<div
  aria-hidden="true"
/>
`;
